securesystemslib/interface.py

```diff
@@ -229,187 +247,191 @@ def generate_and_write_rsa_keypair(filepath=None, bits=DEFAULT_RSA_KEY_BITS,
 
 
 
+def generate_and_write_rsa_keypair(password, filepath=None,
+      bits=DEFAULT_RSA_KEY_BITS):
+  """Generates RSA key pair and writes PEM-encoded keys to disk.
+
+  The private key is encrypted using the best available encryption algorithm
+  chosen by 'pyca/cryptography', which may change over time. The private key is
+  written in PKCS#1 and the public key in X.509 SubjectPublicKeyInfo format.
+
+  NOTE: A signing scheme can be assigned on key import (see import functions).
+
+  Arguments:
+    password: An encryption password.
+    filepath (optional): The path to write the private key to. If not passed,
+        the key is written to CWD using the keyid as filename. The public key
+        is written to the same path as the private key using the suffix '.pub'.
+    bits (optional): The number of bits of the generated RSA key.
+
+  Raises:
+    UnsupportedLibraryError: pyca/cryptography is not available.
+    FormatError: Arguments are malformed.
+    ValueError: An empty string is passed as 'password'.
+    StorageError: Key files cannot be written.
+
+  Side Effects:
+    Writes key files to disk.
+
+  Returns:
+    The private key filepath.
 
-def import_rsa_privatekey_from_file(filepath, password=None,
-    scheme='rsassa-pss-sha256', prompt=False,
-    storage_backend=None):
   """
-  <Purpose>
-    Import the PEM file in 'filepath' containing the private key.
+  securesystemslib.formats.PASSWORD_SCHEMA.check_match(password)
```

Comment:
See my comment above about moving the password schema check to `_get_key_file_encryption_password()`.

Reply:
That's already the case. However, it only checks if the password is not None. That's because `_generate_and_write_*_keypair` (which uses `_get_key_file_encryption_password`) accept None for a password so that callers can indicate the desire to not encrypt. (see docstrings)

`generate_and_write_*_keypair` OTOH do not accept None as password, that's why we unconditionally check the schema here right away.